### PR TITLE
Make a more descriptive functionname and reformulate some documentation 

### PR DESCRIPTION
**Pull request by Jeroenvh99**
@W2Wizard did I do the branching thing correctly? in this PR I've micro-optimized the mlx_equal functions, because the copying of the arguments isn't really necessary, but I'm not really sure if this is actually a good optimization, see my question in discussion #66. my main goal of this PR was to practice with making my edits on a new branch, so if the optimization doesn't have an effect you can just ignore this.

### CODE REVIEW

**Comment by Jeroenvh99** [comment]
I don't know why the build-unix check failed this time tho, did I do something wrong when creating the branch?

**Comment by W2Wizard** [comment]
Fewer lines of code != faster.

In most cases yes, but cramping everything into a single line with a bunch of casts makes no difference to creating two vars instead and makes readability worse. For all I care is that the compiler recognizes what I want and does its best to optimizes it.

**Comment by Jeroenvh99** [comment]
I agree, readability is more important, sorry for this silly thought of me.

File: src/mlx_exit.c [hunk 1]
```diff
@@ -6,7 +6,7 @@
 /*   By: W2Wizard <w2.wizzard@gmail.com>              +#+                     */
 /*                                                   +#+                      */
 /*   Created: 2021/12/28 02:43:22 by W2Wizard      #+#    #+#                 */
-/*   Updated: 2022/11/22 08:59:26 by jvan-hal      ########   odam.nl         */
+/*   Updated: 2022/11/26 14:23:55 by jvan-hal      ########   odam.nl         */
 /*                                                                            */
```
Comment (W2Wizard) [inline src/mlx_exit.c:10]: Not necessary

File: src/utils/mlx_list.c [hunk 2]
```diff
@@ -149,7 +149,7 @@ static void mlx_insertsort(mlx_list_t** head, mlx_list_t* new)
 
 	if (*head == NULL)
 		*head = new;
-	else if (mlx_getdata(*head) >= mlx_getdata(new))
+	else if (mlx_getzdata(*head) >= mlx_getzdata(new))
```
Comment (W2Wizard) [inline src/utils/mlx_list.c:152]: I don't reaaaallly see the benefit, but sure.

**Pull request merged**